libs/designsystem/src/lib/components/chip/chip.component.scss

```diff
@@ -28,21 +30,51 @@
   user-select: none;
 
   :host-context(.kirby-color-brightness-dark) {
-    @include utils.theme-background(transparent, 'white');
-    @include selected-and-hover('white');
+    @include interaction-state.apply-hover($flip: true);
+
+    background-color: transparent;
+    color: utils.get-text-color('white');
+
+    &.is-selected {
+      @include interaction-state.apply-hover('l');
+
+      background-color: utils.get-color('white');
+      color: utils.get-color('white-contrast');
+    }
+  }
+
+  :host-context(.kirby-color-brightness-light) {
+    background-color: transparent;
+    color: utils.get-text-color('black');
+
+    &.is-selected {
+      @include interaction-state.apply-hover('xxl', $flip: true);
+
+      background-color: utils.get-color('black');
+      color: utils.get-color('black-contrast');
+    }
   }
 
   :host-context(kirby-segmented-control.compact) {
-    min-width: 44px;
-    flex-shrink: 0;
-    justify-content: center;
+    background-color: transparent;
+    color: utils.get-text-color('black');
+
+    &.is-selected {
+      @include interaction-state.apply-hover('xl', $flip: true);
+
+      background-color: utils.get-color('black');
+      color: utils.get-color('black-contrast');
+    }
+
     --inline-margin: #{utils.size('xxxxs')};
     --inline-padding: #{13px};
-    @include utils.theme-background(transparent);
-    @include selected-and-hover('black');
 
     @include utils.media('<=small') {
       --inline-margin: 0;
     }
+
+    min-width: 44px;
+    flex-shrink: 0;
+    justify-content: center;
```

Comment:
A bit nitpicky, but consider moving `min-width`, `flex-shrink`, `justify-content`, `--inline-margin` and `--inline-padding` to be before any of mixins and nested selectors in. I think they drown a bit being after all this.

Reply:
That comment made my brain grow a bit! Thanks :D